sbt-dotty/src/dotty/tools/sbtplugin/DottyPlugin.scala

```diff
@@ -174,7 +178,43 @@ object DottyPlugin extends AutoPlugin {
           scalaVersion.value.split("\\.").take(2).mkString(".")
         else
           scalaBinaryVersion.value
-      }
+      },
+
+      scalaInstance := Def.taskDyn {
+        val si = scalaInstance.value
+        if (isDotty.value) {
+          Def.task {
+            val dottydocArtifacts = fetchArtifactsOf("dotty-doc").value
+            val includeArtifact = (f: File) => f.getName.endsWith(".jar")
+            val dottydocJars = dottydocArtifacts.filter(includeArtifact).toArray
+            val allJars = (si.allJars ++ dottydocJars).distinct
+            val loader = new URLClassLoader(Path.toURLs(dottydocJars), si.loader)
+            new ScalaInstance(si.version, loader, si.loaderLibraryOnly, si.libraryJar, si.compilerJar, allJars, si.explicitActual)
+          }
+        } else {
+          Def.task { si }
+        }
+      }.value
     )
   }
+
+  /** Fetch artefacts for scalaOrganization.value %% moduleName % scalaVersion.value */
+  private def fetchArtifactsOf(moduleName: String) = Def.task {
+    val dependencyResolution = Keys.dependencyResolution.value
+    val log = streams.value.log
+    val scalaInfo = scalaModuleInfo.value
+    val updateConfiguration = Keys.updateConfiguration.value
+    val warningConfiguration = (unresolvedWarningConfiguration in update).value
+
+    val moduleID = (scalaOrganization.value %% moduleName % scalaVersion.value).cross(CrossVersion.binary)
+    val descriptor = dependencyResolution.wrapDependencyInModule(moduleID, scalaInfo)
+
+    dependencyResolution.update(descriptor, updateConfiguration, warningConfiguration, log) match {
+      case Left(uw) =>
```

Comment:
Nitpick: since `uw` is unused I would rewrite as:
```scala
... match {
  case Right(report) =>
    report.allFiles
  case _ =>
    throw new MessageOnlyException(...
```